lib/src/executor/runtime_host.rs

```diff
@@ -944,7 +951,7 @@ impl Inner {
                             Ok(())
                         }
                     }
-                    match fmt::write(&mut WriterWithMax(&mut self.logs), format_args!("{req}")) {
+                    match fmt::write(&mut WriterWithMax(&mut self.logs), format_args!("{req}\n")) {
```

Comment:
That doesn't seem right?

```suggestion
                    match fmt::write(&mut WriterWithMax(&mut self.logs), format_args!("{req}")) {
```

A `LogEmit` is emitted when the runtime prints out *something*, not necessarily a line. The runtime has also host functions available that print just a number for example.

Reply:
Yeah but this will merge logs together into one line and sometimes one word without any space

Reply:
That still doesn't make it right. Plus, it's off topic for this PR and should be in a different PR.

Reply:
https://github.com/smol-dot/smoldot/pull/283